bin/sozo/src/commands/events.rs

```diff
@@ -59,59 +63,67 @@ impl EventsArgs {
+
+                let event_filter = EventFilter {
+                    from_block: Some(BlockId::Number(current_start)),
+                    to_block: Some(BlockId::Number(chunk_end)),
+                    address: Some(world_diff.world_info.address),
+                    keys: self.events.as_ref().map(|e| {
+                        vec![e.iter().map(|event| starknet_keccak(event.as_bytes())).collect()]
+                    }),
+                };
 
-            let res =
-                provider.get_events(event_filter, self.continuation_token, self.chunk_size).await?;
-
-            for event in &res.events {
-                match world::Event::try_from(event) {
-                    Ok(ev) => {
-                        match_event(
-                            &ev,
-                            &world_diff,
-                            event.block_number,
-                            event.transaction_hash,
-                            &provider,
-                        )
-                        .await
-                        .unwrap_or_else(|e| {
-                            tracing::error!(?e, "Failed to process event: {:?}", ev);
-                        });
-                    }
-                    Err(e) => {
-                        tracing::error!(
-                            ?e,
-                            "Failed to parse remote world event which is supposed to be valid."
-                        );
+                let res = provider
+                    .get_events(event_filter, continuation_token.clone(), self.chunk_size)
+                    .await?;
+
+                for event in &res.events {
+                    match world::Event::try_from(event) {
+                        Ok(ev) => {
+                            match_event(
+                                &ev,
+                                &world_diff,
+                                event.block_number,
+                                event.transaction_hash,
+                                &provider,
+                            )
+                            .await
+                            .unwrap_or_else(|e| {
+                                tracing::error!(?e, "Failed to process event: {:?}", ev);
+                            });
+                        }
+                        Err(e) => {
+                            tracing::error!(
+                                ?e,
+                                "Failed to parse remote world event which is supposed to be valid."
+                            );
+                        }
                     }
                 }
-            }
 
-            if let Some(continuation_token) = res.continuation_token {
-                println!("Continuation token: {:?}", continuation_token);
-                println!("----------------------------------------------");
+                if let Some(token) = res.continuation_token {
+                    continuation_token = Some(token);
+                } else {
```

Comment:
Here we need the same logic as we have in the `events_to_remote`. Because currently you are not fetching all the pages for the given `EventFilter`. Hence we will loose some events.

You will need to loop while the continuation token is `None`. Then you can update the `current_start`.